narwhals/_arrow/utils.py

```diff
@@ -246,3 +251,95 @@ def cast_for_truediv(arrow_array: Any, pa_object: Any) -> tuple[Any, Any]:
         )
 
     return arrow_array, pa_object
+
+
+def broadcast_series(series: list[ArrowSeries]) -> list[Any]:
+    lengths = [len(s) for s in series]
+    max_length = max(lengths)
+    fast_path = all(_len == max_length for _len in lengths)
+
+    if fast_path:
+        return [s._native_series for s in series]
+
+    pa = get_pyarrow()
+    np = get_numpy()
+
+    reshaped = []
+    for s, length in zip(series, lengths):
+        s_native = s._native_series
+        if max_length > 1 and length == 1:
+            pa_dtype = s_native.type
+            np_dtype = pyarrow_to_numpy_dtype(pa_dtype)
+
+            value = s_native[0]
+
+            if hasattr(value, "as_py"):  # pragma: no cover
+                value = value.as_py()
+
+            reshaped.append(
+                pa.array(
+                    np.full(shape=max_length, fill_value=value, dtype=np_dtype),
+                    type=pa_dtype,
+                )
+            )
+        else:
+            reshaped.append(s_native)
+
+    return reshaped
+
+
+def pyarrow_to_numpy_dtype(pa_dtype: Any) -> Any:  # pragma: no cover
```

Comment:
Shall we test this? 🙈

Reply:
Alright, if this doesn't deserve or require a test, PR should be ready to review

Reply:
Rollback:

![RollbackGIF](https://github.com/user-attachments/assets/d2123295-71f5-4904-99f0-9e2689989049)

Reply:
eventually we can split numeric vs rest